File: client/src/app/core/api/requests/save-profile.request.ts
import { AppTheme } from "../../enums/app-theme.enum";
import { Gender } from "../../enums/gender.enum";
import { Region } from "../../enums/region.enum";

export interface SaveProfileRequest {
  avatarUrl?: string | null;
  firstName?: string | null;
  lastName?: string | null;
  bio?: string | null;
  gender?: Gender | null;
  region?: Region | null;
  theme?: AppTheme | null;
  receiveEmailNotifications?: boolean | null;
}

File: backend/Application/Contracts/Requests/SaveProfileRequest.cs
using CoachApi.Domain.Enums;

namespace CoachApi.Application.Contracts.Requests;

public class SaveProfileRequest
{
    public string? AvatarUrl { get; set; }
    public string? FirstName { get; set; }
    public string? LastName { get; set; }
    public string? Bio { get; set; }
    public Gender? Gender { get; set; }
    public Region? Region { get; set; }
    public AppTheme? Theme { get; set; }
    public bool? ReceiveEmailNotifications { get; set; }
}

